src/pip/_internal/operations/prepare.py

```diff
@@ -585,14 +562,13 @@ def prepare_editable_requirement(
                     'hash.'.format(req)
                 )
             req.ensure_has_source_dir(self.src_dir)
-            req.update_editable(not self._download_should_save)
+            req.update_editable(self.download_dir is None)
 
             dist = _get_prepared_distribution(
                 req, self.req_tracker, self.finder, self.build_isolation,
             )
 
-            if self._download_should_save:
-                req.archive(self.download_dir)
+            req.archive(self.download_dir)
```

Comment:
This doesn't look right. Is this covered by the build_dir changes?

Reply:
It is covered by the change to `req.archive`, since more than one call to it check for the same condition: https://github.com/pypa/pip/pull/8843/commits/ceaddcc76bb6c96af0f12167e9d692437eb44b7e